### PR TITLE
Milestone logic

### PR DESCRIPTION
**Pull request by HADDADSOHAIB**
I added the necessary logic elements for the working of the calculator:
- Operate function that does the math.
- Calculate function that manages the workflow.

### CODE REVIEW

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @HADDADSOHAIB,
You have done a great job in this milestone, however, just a couple of things need to be changed for this to be approved.

**See my comments below for a review**

### Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions_

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    };
+  }
+
+  if (buttonName === '.' && next && !next.includes('.')) {
+    return {
+      total,
+      next: `${next}.`,
+      operation,
+    };
+  }
```
Comment (fatymahmed) [inline src/logic/calculate.js]: - When the `buttonName` is `.` there are many more cases that you should consider.
- [x] What if `next` is `null`, but `operation` is not `null` and the user clicks `.` ? (this should set `next` to `0.`)
- [x] What is `next` is `null` and `total` isn't null and the user clicks `.` ? (`.` should be included in `total`)
- [ ] What if `next` exists and `.` is already included in it but user clicks on `.` again ?

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  if (buttonName.match(/[-/X+]/) && next && !total) {
+    return {
+      total: next,
+      next: null,
+      operation: buttonName,
+    };
+  }
```
Comment (fatymahmed) [inline src/logic/calculate.js]: Make sure to use what you imported from `operate.js` which covered these cases

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return oneBig.mul(twoBig);
+  }
+
+  if (op === '/' && two !== 0) {
+    return oneBig.div(twoBig).toFixed(2);
+  }
+
+  if (op === '/' && two === 0) {
+    return Infinity;
+  }
```
Comment (fatymahmed) [inline src/logic/operate.js]: Remember that the `result` prop in `Display` component should be a string so maybe you could convert these to strings here.

File: src/logic/calculate.js (hunk 1 above)
Comment (HADDADSOHAIB) [reply src/logic/calculate.js]: I believe in the third case nothing will be changed (. already in next), so this case is handled by the last return,
```
 return {
      total,
      next,
      operation,
  };
```

File: src/logic/calculate.js (hunk 2 above)
Comment (HADDADSOHAIB) [reply src/logic/calculate.js]: Hi @fatymahmed, I am using operate function (what imported from operate.js), so your comment is not clear. An additional explanation is required. Thanks :)

File: src/logic/calculate.js (hunk 1 above)
Comment (fatymahmed) [reply src/logic/calculate.js]: You are right

> I believe in the third case nothing will be changed (. already in next), so this case is handled by the last return,
> 
> ```
>  return {
>       total,
>       next,
>       operation,
>   };
> ```

File: src/logic/calculate.js (hunk 2 above)
Comment (fatymahmed) [reply src/logic/calculate.js]: Hi @HADDADSOHAIB, you are right, you are already using the `operate` from `operate.js`, my apologies, I missed that.

File: src/logic/calculate.js [hunk 4]
```diff
@@ -76,14 +100,14 @@ const calculate = ({ total, next, operation }, buttonName) => {
   if (buttonName === '%' && next && total) {
     return {
       total,
-      next: (operate(parseFloat(next), 100, '/')).toString(),
+      next: operate(parseFloat(next), 100, '/'),
```
Comment (MauricioRobayo) [inline src/logic/calculate.js:103]: I like the way you did this 👍 .

**Review by MauricioRobayo — APPROVED** [review]
Great, looking good so far.

Thanks for fixing the issues 😄 .

Status: __Approved__.

**Pull request closed without merging**